### PR TITLE
Displayed image and added mouse and key listener

### PR DESCRIPTION
**Pull request by keldavis**

### CODE REVIEW

File: web/server.py [hunk 1]
```diff
@@ -1,54 +1,72 @@
...
+    def connected(self):
+        """
+        Display connected address and send image path
+        """
+        print(self.address, 'connected')
+        self.send_image_path()
+
+    def handle_close(self):
+        """
+        Display closed address and send image path
```
Comment (daniel-falk) [inline web/server.py:53]: Incorrect comment: no image path sent in handle_close

File: web/test.html [hunk 2]
```diff
@@ -1,110 +1,210 @@
...
+
+
+<script language="javascript" type="text/javascript">
+
+  //add event listener for page load
+  window.addEventListener("load", init, false);
+
+  //add event listener for key press
+  document.addEventListener('keyup', (e) => {
+    if (e.code === "ArrowLeft") {
```
Comment (daniel-falk) [inline web/test.html:44]: Can we change line 44-50 to simply forward all key events, then we can filter in the back end instead. We could also try to be consistent in the format with the mouse click events:

doSend("EVENT(key)=" + e.code)

File: web/test.html [hunk 3]
```diff
@@ -1,110 +1,210 @@
...
+    }
+  }
+
+  //Function to get normalize coordinates of image
+  function GetImageCoordinates(evt) {
+    var x = 0;
+    var y = 0;
+    var imgPos;
+    imgPos = findImagePosition(img);
+    if (!evt) var evt = window.event;
```
Comment (daniel-falk) [inline web/test.html:149]: A comment would be nice explaining when evt is injected as the argument and when we need to get is from the window object

File: web/test.html [hunk 4]
```diff
@@ -1,110 +1,210 @@
...
+    if (evt.pageX || evt.pageY) {
```
Comment (daniel-falk) [inline web/test.html:150]: Add a comment explaining why we need to check both pageX and clientX, browser dependent or dependent on what we click on?

File: web/test.html [hunk 5]
```diff
@@ -1,110 +1,210 @@
...
+
+    y -= imgPos[1]
+    if (y < 0) {
+      y = 0;
+    }
+    else if (y > img.height) {
+      y = img.height
+    }
+    y /= img.height;
+    doSend("COORDINATES(x,y)=(" + x + "," + y+")")
```
Comment (daniel-falk) [inline web/test.html:178]: considering the function name "getImageCoordinates" it is a bit confusing that it does not return anything but instead sends the result to the back end. Perhaps we could add a new function:

function sendImageClickCoordinates(evt) {
    var coords = getImageCoordinates(evt)
    doSend("COORDINATES(x,y)=(" + coords.x + "," + coords.y+")")
}

File: web/test.html [hunk 6]
```diff
@@ -1,110 +1,210 @@
...
+      <input type="button" name=sendButton value="Send" onClick="sendText();">
+      <input type="button" name=clearButton value="Clear" onClick="clearText();">
+      <input type="button" name=disconnectButton value="Disconnect" onClick="doDisconnect();">
+      <input type="button" name=connectButton value="Connect" onClick="doConnect();">
+      </p>
+    </div>
+  </div>
+</form>
+<script>
+  //Getting reference of image
```
Comment (daniel-falk) [inline web/test.html:204]: This comment is a bit redundant - comments should normally tell why you do something, not what you do (that's clear from the code itself). E.g.:

// Attach mouse event to image to send the clicked coordinates to the back end
var img = document.getElementById("image");
img.onmousedown = GetImageCoordinates;

**Review by daniel-falk — CHANGES_REQUESTED** [review]
I like the result, it works great. See comments for some minor things.

**Comment by keldavis** [comment]
Hi,

I sent the pull request a while back. 

Kelli Davis

**Review by daniel-falk — APPROVED** [review]
Excellent, I'll merge this to the feature branch. Thanks Kelli!

**Pull request merged**